include/openthread/dataset.h

```diff
@@ -99,23 +99,13 @@ struct otExtendedPanId
  */
 typedef struct otExtendedPanId otExtendedPanId;
 
-#define OT_MESH_LOCAL_PREFIX_SIZE 8 ///< Size of the Mesh Local Prefix (bytes)
+#define OT_MESH_LOCAL_PREFIX_SIZE OT_IP6_PREFIX_SIZE ///< Size of the Mesh Local Prefix (bytes)
 
 /**
  * This structure represents a Mesh Local Prefix.
  *
  */
-OT_TOOL_PACKED_BEGIN
-struct otMeshLocalPrefix
-{
-    uint8_t m8[OT_MESH_LOCAL_PREFIX_SIZE]; ///< Byte values
-} OT_TOOL_PACKED_END;
-
-/**
- * This structure represents a Mesh Local Prefix.
- *
- */
-typedef struct otMeshLocalPrefix otMeshLocalPrefix;
+typedef otIp6NetworkPrefix otMeshLocalPrefix;
```

Comment:
Should we replace `otMeshLocalPrefix` with `otIp6NetworkPrefix`?

Reply:
I think it's a good idea to remove the public  `otMeshLocalPrefix` type definition itself (and use `otIp6NetworkPrefix` in its place directly). 

What do you think about keeping or removing `Mle::MeshLocalPrefix` class?
- I favor keeping it.
- With this PR, it inherits from `Ip6::NetworkPrefix` and adds one method `SetFromExtendedPanId()`.
- We can consider moving this method to `NetworkPrefix` but I feel it's  MLE specific.

Reply:
I'm fine with keeping `Mle::MeshLocalPrefix` for the reasons you mentioned.

Reply:
Updated in the new push. 

Also added a `GenerateRandomUla()` method to generate a ULA based random prefix of `fdxx:xxxx:xxxx:xxxx:` pattern.

Reply:
Per offline chat with @jwhui , decided to keep the public `otMeshLocalPrefix` definition (we define it as a `typedef` of the `otIp6NetworkPrefix`).